packages/graphiql-react/src/translation/translation-provider.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState, useEffect } from 'react';
+import { TranslationService } from './services/translation-service';
+import { TranslationContext } from './translation-context';
+import { Language } from './models/language';
+
+import en from '../assets/i18n/en.json';
+import fr from '../assets/i18n/fr.json';
+import { TranslationBundle } from './models/translation-bundle';
+import { Translations } from './models/translations';
+
+export const TranslationProvider = (props: any) => {
+  const allSupportedLanguages = getSupportedLanguages(props.languages);
+  const selectedLanguage = allSupportedLanguages.find(
+    language => language.code === props.selectedLanguage,
+  );
+
+  const translationService = new TranslationService(
+    getTranslationBundles(props.translations),
+  );
+
+  const [currentLanguage, setCurrentLanguage] = useState<Language>(
+    selectedLanguage || Language.EN,
+  );
+  const [supportedLanguages, setSupportedLanguages] = useState<Language[]>(
+    allSupportedLanguages,
+  );
+
+  useEffect(() => {
+    if (selectedLanguage && selectedLanguage !== currentLanguage) {
+      setCurrentLanguage(selectedLanguage);
+    }
+  }, [selectedLanguage]);
+
+  return (
+    <TranslationContext.Provider
+      value={{
+        currentLanguage,
+        supportedLanguages,
+        setCurrentLanguage,
+        setSupportedLanguages,
+        translationService,
+      }}
+    >
+      {props.children}
+    </TranslationContext.Provider>
+  );
+};
+
+const getSupportedLanguages = (languages?: Language[]) => {
+  const supportedLanguages: Language[] = [];
+  supportedLanguages.push(Language.EN, Language.FR);
+
+  if (!languages) {
+    return supportedLanguages;
+  }
+
+  for (const language of languages) {
```

Comment:
Why do we need this languages configuration? Can't we just get what's the language from the bundle itself?

Reply:
done